test/integration/component/maint/testpath_disable_enable_host.py

```diff
@@ -0,0 +1,309 @@
+            )
+
+        except Exception as e:
+            cls.tearDownClass()
+            raise e
+        return
+
+    @classmethod
+    def tearDownClass(cls):
+        try:
+            pass
+            cleanup_resources(cls.apiclient, cls._cleanup)
+        except Exception as e:
+            raise Exception("Warning: Exception during cleanup : %s" % e)
+
+    def setUp(self):
+        self.apiclient = self.testClient.getApiClient()
+        self.dbclient = self.testClient.getDbConnection()
+        self.cleanup = []
+
+    def tearDown(self):
+        try:
+            cleanup_resources(self.apiclient, self.cleanup)
+        except Exception as e:
+            raise Exception("Warning: Exception during cleanup : %s" % e)
+        return
+
+    @attr(tags=["advanced", "basic"], required_hardware="true")
+    def test_01_disable_enable_host(self):
+        """disable enable host
+            1. Disable host and verify following things:
+                For admin user:
+                    1. Should be create to start/stop exsiting vms
+                    2. Should be create to deploy new vm,
+                    snapshot on the same host
+                For Non-admin user:
+                    1. Should be create to start/stop exsiting vms
+                    2. Should not be create to deploy new vm,
+                    snapshot on the same host
+            2. Enable the above disabled host and verify that:
+                -All users should be create to deploy new vm,
+                snapshot on the same host
+            3. Try to reconnect the host :
+                -Host should get reconnected successfully
+        """
+        # Step 1
+        vm_root = VirtualMachine.create(
+            self.apiclient,
+            self.testdata["small"],
+            templateid=self.template.id,
+            accountid=self.account.name,
+            domainid=self.account.domainid,
+            serviceofferingid=self.service_offering.id,
+            zoneid=self.zone.id,
+        )
+        hostid = vm_root.hostid
+
+        vm_user = VirtualMachine.create(
+            self.userapiclient,
+            self.testdata["small"],
+            templateid=self.template.id,
+            accountid=self.account.name,
+            domainid=self.account.domainid,
+            serviceofferingid=self.service_offering.id,
+            zoneid=self.zone.id,
+        )
+
+        cmd = updateHost.updateHostCmd()
+        cmd.id = hostid
+        cmd.resourcestate = "Disabled"
+        cmd.allocationstate = "Disable"
+        self.apiclient.updateHost(cmd)
+        hostList = Host.list(self.apiclient, id=hostid)
+
+        self.assertEqual(hostList[0].resourcestate,
+                         "Disabled",
+                         "Check if the host is in disabled state"
+                         )
+        # Verify the exsisting vms should be running
+        self.assertEqual(vm_user.state,
+                         "Running",
```

Comment:
Use constant string from codes.py and also compare string by converting both to lowercase. Change everywhere applicable